### ISSUE TITLE
ManifoldSAE.fit() joint solve broken across configs: ndarray broadcast panic [N*D,M]→[N,M] (+ product 'primary penalty was not built')

### ISSUE THREAD

**Issue by SauersML** [SauersML/gam#356]
## Summary
`gamfit.torch.ManifoldSAE.fit()` — the joint Arrow-Schur co-fit (the headline capability: co-optimize decoder blocks `B_k`, per-token coordinates `t`, assignments, and ρ) — is **broken across configs** in 0.1.134. It either panics in Rust (`ndarray` broadcast) or raises a penalty-build `ValueError`. The forward/backprop path works fine; only the closed-form joint solve is affected. Notably, no torch test appears to exercise `.fit()`, so this shipped silently.

## Repro (gamfit 0.1.134, PyPI wheel)
```python
import torch
from gamfit.torch import ManifoldSAE, ManifoldSAEConfig, SparsityConfig
cfg = ManifoldSAEConfig(input_dim=12, n_atoms=8, n_basis_per_atom=6, intrinsic_rank=1,
                        atom_manifold="circle",
                        sparsity=SparsityConfig(kind="ibp_gumbel", target_k=3),
                        dtype=torch.float64)
sae = ManifoldSAE(cfg)
sae.fit(torch.randn(48, 12, dtype=torch.float64))
```
```
thread '<unnamed>' panicked at ndarray-0.17.2/src/lib.rs:1801:
ndarray: could not broadcast array from shape: [576, 36] to: [48, 36]      # 576 = 48*12 (N*D)
PanicException: ndarray: could not broadcast array from shape: [576, 36] to: [48, 36]
```

## Config matrix (N=48, D=12, n_atoms=8, n_basis_per_atom=6)
| atom_manifold | intrinsic_rank | sparsity | result |
|---|---|---|---|
| circle  | 1 | ibp_gumbel  | **PanicException** broadcast `[576,36] -> [48,36]` |
| circle  | 1 | softmax_topk| **PanicException** broadcast `[576,36] -> [48,36]` |
| sphere  | 2 | jumprelu    | **PanicException** broadcast `[576,84] -> [48,84]` |
| product | 2 | ibp_gumbel  | **ValueError**: `sae_build_duchon_atom: primary penalty was not built` |

The `[N*D, M] -> [N, M]` shape strongly suggests the solver assembles a per-(token×output-channel) design/basis matrix (`N*D` rows) somewhere it should be per-token (`N` rows) — i.e. the multi-output (D-dim ambient) reshaping is wrong. The `product` case fails earlier, in the Duchon atom penalty assembly.

## Impact
The joint co-fit is the core differentiator over a plain backprop autoencoder (co-optimizing coordinates + shape + assignments + ρ, vs treating the curve as a frozen byproduct). With `.fit()` unusable, downstream is forced onto pure end-to-end backprop, losing the capability. A Rust `panic!` also aborts hard (PanicException) rather than a catchable error.

## Also observed
`[GPU] CUDA acceleration disabled: CUDA support not compiled into this build` — the fit runs CPU-only in this wheel (separate issue; relevant for LLM-scale).

**Comment by SauersML** [comment]
Closing — fix landed in fc6b16088 and now has a real regression test (5aa24c4d4).

**Root-cause fix (fc6b16088), both failure modes:**

*Part 1 — broadcast panic `[N*D,M]→[N,M]` (circle/sphere/cylinder rows).* The SAE pre-fit decoder identifiability audit was structurally mis-specified: a single p-vector predictor per row has decoder Jacobian `(n·p, M_k·p)`, which the cross-block channel-aware/flat audit conflated with the `(n)`-row placeholder design and broadcast `(n·p, …)` into `(n, …)`. Since `H_data = G_k ⊗ I_p`, decoder identifiability is fully determined by the per-atom `(n, M_k)` weighted design `D_k = diag(a_·k)·Φ_k` — the p-fold output replication carries no extra structural information. The audit now runs a per-atom rank check directly on `D_k` (via the Gram `G_k = D_kᵀD_k`, chunk-aware), never materialising the `(n·p, …)` block. Rank-0 → hard error; rank-deficient → INFO + ridge regularises. The dead `decoder_parameter_block_specs` + `SaeDecoderBlockJacobian` machinery and the unsound flat-audit instrumentation call in `canonicalize_for_identifiability` were deleted (src/terms/sae_manifold.rs:3365-3394, 3630-3668).

*Part 2 — `sae_build_duchon_atom: primary penalty was not built` (product/Duchon row).* The scale-free pure-Duchon path emits the curvature seminorm as `OperatorStiffness`, never `Primary`; the builder searched for `Primary` and resolved a non-zero power inconsistent with the power-0 `DuchonCoordinateEvaluator`. It now builds the penalty exactly like `duchon_function_norm_penalty` (power 0, stiffness-only, dimension-aware `m = floor(dim/2)+2`).

**Regression coverage:**
- New `tests/torch/test_manifold_sae_fit_config_matrix_issue_356.py` exercises the exact repro: `ManifoldSAE.fit()` at N=48, D=12, n_atoms=8, n_basis_per_atom=6 across all four config-matrix rows (circle+ibp_gumbel, circle+softmax_topk, sphere+jumprelu, product+ibp_gumbel) — each must run the joint solve to completion with finite decoder blocks / reconstruction / assignments. Directly addresses the issue's note that no torch test exercised `.fit()`.
- `tests/sae_audit_is_invoked.rs` (updated by the fix) pins the new per-atom rank audit (well-posed passes; rank-0 atom surfaces an identifiability error pre-Newton).
- `tests/test_sae_manifold_duchon_euclidean_issue.py` pins the Duchon 1D 'primary penalty was not built' and 2D collocation paths (Part 2).

Principled root-cause repair, no shim/back-compat, dead machinery removed, edge cases handled. Meets the bar.
